File: task-03/C++/Sub task 3/subtask3.cpp
#include <iostream>

void printDiamond(int n) {
    int i, j;

    // Upper part of the diamond
    for (i = 1; i <= n; i += 2) {
        // Print leading spaces
        for (j = (n - i) / 2; j > 0; j--) {
            std::cout << " ";
        }
        // Print stars
        for (j = 0; j < i; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }

    // Lower part of the diamond
    for (i = n - 2; i >= 1; i -= 2) {
        // Print leading spaces
        for (j = (n - i) / 2; j > 0; j--) {
            std::cout << " ";
        }
        // Print stars
        for (j = 0; j < i; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }
}

int main() {
    int n;

    std::cout << "Enter the number of rows for the diamond: ";
    std::cin >> n;

    if (n < 1) {
        std::cout << "Number should be greater than 1\n";
        return 1;
    }

    printDiamond(n);

    return 0;
}

File: task-03/Java/Sub task 3/subtask3.java
import java.util.Scanner;

public class subtask3 {

    public static void printDiamond(int n) {
        if (n < 1) {
            System.out.println("Number should be greater than 1");
            return;
        }

        // Calculate the middle point
        int mid = n / 2;

        // Upper part of the diamond
        for (int i = 1; i <= n; i += 2) {
            // Calculate and print leading spaces
            int spaces = mid - i / 2;
            System.out.print(" ".repeat(spaces));

            // Print stars
            System.out.println("*".repeat(i));
        }

        // Lower part of the diamond
        if (n % 2 == 0) {
            for (int i = n - 1; i >= 1; i -= 2) {
                // Calculate and print leading spaces with a slight shift for asymmetry
                int spaces = mid - i / 2 + 1;  // Slightly shift the spaces to the right
                System.out.print(" ".repeat(spaces - 1));  // Reduce one space for left shift

                // Print stars
                System.out.println("*".repeat(i));
            }
        } else {
            for (int i = n - 2; i >= 1; i -= 2) {
                // Calculate and print leading spaces
                int spaces = mid - i / 2;
                System.out.print(" ".repeat(spaces));

                // Print stars
                System.out.println("*".repeat(i));
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of rows for the diamond: ");
        int n = scanner.nextInt();

        printDiamond(n);
    }
}

